File: 1-Two-Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        freq = {}

        for i in range(len(nums)):
            dif = target - nums[i]
            if dif in freq:
                return [nums.index(dif), i]
            freq[nums[i]] = dif

        return {-1}

File: 1-Two-Sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)

    for i, num := range nums {
        complement := target - num
        if index, ok := m[complement]; ok {
            return []int{index, i}
        }
        m[num] = i
    }

    return nil
}
